pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/DistinctCountThetaSketchAggregationFunction.java

```diff
@@ -83,16 +73,69 @@
    *                    <li> Required: Last expression is the one that will be evaluated to compute final result. </li>
    *                    </ul>
    */
-  public DistinctCountThetaSketchAggregationFunction(List<String> arguments)
+  public DistinctCountThetaSketchAggregationFunction(List<ExpressionContext> arguments)
       throws SqlParseException {
-    int numExpressions = arguments.size();
+    int numArguments = arguments.size();
+
+    // NOTE: This function expects at least 3 arguments: theta-sketch column, parameters, post-aggregation expression.
+    Preconditions.checkArgument(numArguments >= 3,
+        "DistinctCountThetaSketch expects at least three arguments (theta-sketch column, parameters, post-aggregation expression), got: ",
+        numArguments);
+
+    // Initialize the theta-sketch column
+    _thetaSketchColumn = arguments.get(0);
+    Preconditions.checkArgument(_thetaSketchColumn.getType() == ExpressionContext.Type.IDENTIFIER,
+        "First argument of DistinctCountThetaSketch must be identifier (theta-sketch column)");
+
+    // Initialize the theta-sketch parameters
+    ExpressionContext paramsExpression = arguments.get(1);
+    Preconditions.checkArgument(paramsExpression.getType() == ExpressionContext.Type.LITERAL,
+        "Second argument of DistinctCountThetaSketch must be literal (parameters)");
+    _thetaSketchParams = ThetaSketchParams.fromString(paramsExpression.getLiteral());
 
-    // This function expects at least 3 arguments: Theta Sketch Column, Predicates & final aggregation expression.
-    Preconditions.checkArgument(numExpressions >= 3, "DistinctCountThetaSketch expects at least three arguments, got: ",
-        numExpressions);
+    // Initialize the theta-sketch set operation builder
+    _setOperationBuilder = getSetOperationBuilder();
 
-    // Initialize all the internal state.
-    init(arguments);
+    // Initialize the input expressions
```

Comment:
what's the issue with calling init()?

Reply:
With `init()` the member variables cannot be declared as `final`. IMO declaring them as `final` can enhance the readability because these variables should only be set once and should not be changed.

Reply:
👍